### ISSUE TITLE
Comparison of String and dynamic (function) fails with uncaught error in core_ajax_dart.dart

### ISSUE THREAD

**Issue by DartBot** [dart-lang/polymer-dart#301]
_Originally opened as dart-lang/sdk#20978_

*This issue was originally filed by frable9...&#064;gmail.com*
_____
**What steps will reproduce the problem?**
1. Fire a CoreAjax request with go()

**What is the expected output? What do you see instead?**
I get an uncaught error from the core_ajax_dart.dart.
The essential code snippet from core_ajax_dart.dart is:
if (!hasContentType &amp;&amp; this.contentType) {
&nbsp;&nbsp;headers['Content-Type'] = this.contentType;
}
whereby hasContentType is a function with a boolean as return value and this.contentType is a String.

Exception: Uncaught Error: type 'String' is not a subtype of type 'bool' of 'boolean expression'.
Stack Trace:
#­0      CoreAjax.go (package:core_elements/core_ajax_dart.dart:285:33)
#­1      AelAjax.go (http://localhost:8080/components/ael-ajax/ael-ajax.dart:39:17)
#­2      AelCtrl.getUserProfile (http://localhost:8080/app.dart:157:26)
#­3      AelCtrl.AelCtrl (http://localhost:8080/app.dart:77:19)
#­4      main.&lt;anonymous closure&gt;.&lt;anonymous closure&gt; (http://localhost:8080/app.dart:27:18)
#­5      \_RootZone.runUnary (dart:async/zone.dart:1082)
#­6      \_Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:488)
#­7      \_Future._propagateToListeners (dart:async/future_impl.dart:571)
#­8      \_Future._completeWithValue (dart:async/future_impl.dart:331)
#­9      \_Future._asyncComplete.&lt;anonymous closure&gt; (dart:async/future_impl.dart:393)
#­10     \_asyncRunCallbackLoop (dart:async/schedule_microtask.dart:41)
#­11     \_asyncRunCallback (dart:async/schedule_microtask.dart:48)
#­12     \_handleMutation (dart:html:39006)

**What version of the product are you using?**
core_elements 0.2.1+1
Dart 1.6.0

**On what operating system?**
Windows 7 64 bit

**What browser (if applicable)?**
Dartium 37.0.2062.76

**Please provide any additional information below.**

**Comment by DartBot** [comment]
<a href="https://github.com/iposva-google"><img src="https://avatars.githubusercontent.com/u/5449880?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [iposva-google](https://github.com/iposva-google)**

----

*Added [Pkg-Polymer](../labels/Pkg-Polymer), [Area-Pkg](../labels/Area-Pkg) labels.*

**Comment by DartBot** [comment]
<a href="https://github.com/iposva-google"><img src="https://avatars.githubusercontent.com/u/5449880?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [iposva-google](https://github.com/iposva-google)**

----

*Added [Triaged](../labels/Triaged) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/jakemac53"><img src="https://avatars.githubusercontent.com/u/984921?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [jakemac53](https://github.com/jakemac53)**

----

Duplicate of https://github.com/dart-lang/core-elements/issues/107
______
*Set owner to @jakemac53.*
*Removed [Priority-Unassigned](../labels/Priority-Unassigned) label.*
*Added [Priority-High](../labels/Priority-High), [MovedToGithub](../labels/MovedToGithub) labels.*
